pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -335,15 +335,22 @@ impl InMemoryLayer {
                     lsn
                 );
                 inner.segsizes.insert(lsn, newsize);
+                // ensure there is no gaps when segment is enlarged
```

Comment:
I'm not sure that's a safe assumption. PostgreSQL can extend a relation concurrently in two different backends, but WAL-log the new pages in different order. For example, if you have two backends inserting into a heap at the same time, backend A might get block 100 and backend B block 101. But backend B might WAL-log block 101 before A logs block 100.

We currently do extra WAL-logging on every page extension in Zenith, which might make that impossible, I'm not sure. See https://github.com/zenithdb/zenith/issues/482. But we should stop doing that for performance reasons anyway.

Reply:
Yeah, this assumption was wrong. CI also pointed this out in pgbench test case:
```
thread 'WAL receiver thread' panicked at 'assertion failed: newsize - oldsize <= 1'
```